nucypher/cli/main.py

```diff
@@ -242,17 +254,71 @@ def ursula(click_config,
+            how_to_run_message = "\nTo run an Ursula node from the default configuration filepath run: \n\n'{}'\n"
+            suggested_command = 'nucypher ursula run'
+            if config_root is not None:
+                config_file_location = os.path.join(config_root, config_file or UrsulaConfiguration.CONFIG_FILENAME)
+                suggested_command += ' --config-file {}'.format(config_file_location)
+            click.secho(how_to_run_message.format(suggested_command), fg='green')
+            return  # FIN
+
+        else:
+            click.secho("OK")
+
+    elif action == "destroy":
+        """Delete all configuration files from the disk"""
 
-        # Give the use a suggestion as to what to do next...
-        how_to_run_message = "\nTo run an Ursula node from the default configuration filepath run: \n\n'{}'\n"
-        suggested_command = 'nucypher ursula run'
-        if config_root is not None:
-            config_file_location = os.path.join(config_root, config_file or UrsulaConfiguration.CONFIG_FILENAME)
-            suggested_command += ' --config-file {}'.format(config_file_location)
-        click.secho(how_to_run_message.format(suggested_command), fg='green')
-        return  # FIN
+        if dev:
+            message = "'nucypher ursula destroy' cannot be used in --dev mode"
+            raise click.BadOptionUsage(option_name='--dev', message=message)
+
+        try:
+            ursula_config = UrsulaConfiguration.from_configuration_file(filepath=config_file)
+
+        except FileNotFoundError:
+            config_root = config_root or DEFAULT_CONFIG_ROOT
+            config_file_location = config_file or UrsulaConfiguration.DEFAULT_CONFIG_FILE_LOCATION
+
+            if not force:
+                message = "No configuration file found at {}; \n" \
+                          "Destroy top-level configuration directory: {}?".format(config_file_location, config_root)
+                click.confirm(message, abort=True)  # ABORT
+
+            shutil.rmtree(config_root, ignore_errors=False)
+
+        else:
+            if not force:
+                click.confirm('''
+*Permanently and irreversibly delete all* nucypher files including
+    - Private and Public Keys
+    - Known Nodes
+    - TLS certificates
+    - Node Configurations
+    - Log Files
+
+Delete {}?'''.format(ursula_config.config_root), abort=True)
+
+            try:
+                ursula_config.destroy(force=force)
+            except FileNotFoundError:
+                message = 'Failed: No nucypher files found at {}'.format(ursula_config.config_root)
+                click.secho(message, fg='red')
+                log.debug(message)
+                raise click.Abort()
+            else:
+                message = "Deleted configuration files at {}".format(ursula_config.config_root)
+                click.secho(message, fg='green')
+                log.debug(message)
+
+        if not quiet:
```

Comment:
Is there not a better way to echo without checking if we're `quiet` each and every time?

Reply:
@tuxxy - In this particular case case - I see what you mean - we can wrap `click.secho` and apply some sort of filtration, for example.

Reply:
Yes, for example having a callback for `--quiet` which sets an option for this wrapper to not `secho` if it is `True`

Reply:
Or even patching `secho` depending on the flag.

But yeah, one way or another, the quiet checks and resulting indentation make it much harder to quickly read and review this part.